backend/app/interfaces/email_service_provider.py

```diff
@@ -8,9 +10,7 @@ class IEmailServiceProvider(ABC):
     """
 
     @abstractmethod
-    def send_email(
-        self, recipient: str, subject: str, body_html: str, body_text: str
-    ) -> dict:
+    def send_email(self, template: EmailTemplate, content: EmailContent) -> dict:
```

Comment:
Ditto `template` vs `templateType`, as template could be a full formed template class.